src/RequestBuilder.php

```diff
@@ -201,7 +201,19 @@ private function constructBodyAndQueryParameters(Message $message, array $config
                 $requiredQueryParam = Serializer::toCamelCase($requiredQueryParam);
                 if (!array_key_exists($requiredQueryParam, $queryParams)) {
                     $getter = Serializer::getGetter($requiredQueryParam);
-                    $queryParams[$requiredQueryParam] = $message->$getter();
+                    $queryParamValue = $message->$getter();
+                    if ($queryParamValue instanceof Message) {
+                        $param = json_decode($queryParamValue->serializeToJsonString(), true);
+                        if (is_array($param)) {
+                            foreach ($param as $key => $value) {
+                                $queryParams[$requiredQueryParam . '.' . $key] = $value;
+                            }
+                        } else {
+                            $queryParams[$requiredQueryParam] = $param;
+                        }
+                    } else {
+                        $queryParams[$requiredQueryParam] = $queryParamValue;
+                    }
```

Comment:
nit: I really want to optimize this code some how because they are the basically the same statement...but I don't have any clever ideas...do you?

Reply:
We could use `continue`:

```php
            foreach ($config['queryParams'] as $requiredQueryParam) {
                $requiredQueryParam = Serializer::toCamelCase($requiredQueryParam);
                if (!array_key_exists($requiredQueryParam, $queryParams)) {
                    $getter = Serializer::getGetter($requiredQueryParam);
                    $queryParamValue = $message->$getter();
                    if ($queryParamValue instanceof Message) {
                        // Decode message for the query parameter.
                        $queryParamValue = json_decode($queryParamValue->serializeToJsonString(), true);
                        if (is_array($queryParamValue)) {
                            // If the message has properties, add them as nested querystring values.
                            // NOTE: If the nested property is also an object, it will be a JSON
                            // object. This means the nesting only happens at one level of depth.
                            foreach ($queryParamValue as $key => $value) {
                                $queryParams[$requiredQueryParam . '.' . $key] = $value;
                            }
                            continue;
                        }
                    }
                    $queryParams[$requiredQueryParam] = $queryParamValue;
                }
            }
```

Reply:
oo yeah that great :)

Reply:
Or split it up like this, which would allow for arrays to be passed in `$config['queryParams']`:
```php
            foreach ($config['queryParams'] as $requiredQueryParam) {
                $requiredQueryParam = Serializer::toCamelCase($requiredQueryParam);
                if (!array_key_exists($requiredQueryParam, $queryParams)) {
                    $getter = Serializer::getGetter($requiredQueryParam);
                    $queryParamValue = $message->$getter();
                    if ($queryParamValue instanceof Message) {
                        // Decode message for the query parameter.
                        $queryParamValue = json_decode($queryParamValue->serializeToJsonString(), true);
                    }
                    if (is_array($queryParamValue)) {
                        // If the message has properties, add them as nested querystring values.
                        // NOTE: If the nested property is also an object, it will be a JSON
                        // object. This means the nesting only happens at one level of depth.
                        foreach ($queryParamValue as $key => $value) {
                            $queryParams[$requiredQueryParam . '.' . $key] = $value;
                        }
                    } else {
                        $queryParams[$requiredQueryParam] = $queryParamValue;
                    }
                }
            }
```